File: Domain/Models/Sector.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace Domain.Models
{
    public class Sector
    {
        public Sector()
        {
            Id = Guid.NewGuid();
            ServiceOwners = new List<ServiceOwner>();
            Services = new List<Service>();
            CategoryServices = new List<CategoryService>();
        }

        public Guid Id { get; set; }
        public string TypeSector { get; set; }
        public string Description { get; set; }
        public string UrlSectorIcon { get; set; }
        public bool IsDeleted { get; set; } = false;
        public bool IsAccepted { get; set; } = false;
        public ICollection<ServiceOwner> ServiceOwners { get; set; }
        public ICollection<Service> Services { get; set; }
        public ICollection<CategoryService> CategoryServices { get; set; }
    }
}

File: ScheduledServicesCenterUI/src/models/sector.ts
import { Guid } from 'guid-typescript';
import { ServiceOwner } from './service.owner';
import { Service } from './service';
import { CategoryService } from './category.service';

export class Sector {
  Id: Guid | undefined;
  TypeSector: string | undefined;
  Description: string | undefined;
  UrlSectorIcon: string | undefined;
  IsAccepted: boolean | undefined;
  IsDeleted: boolean | undefined;
  ServiceOwners: ServiceOwner[] | undefined;
  Services: Service[] | undefined;
  CategoryServices: CategoryService[] | undefined;
}
